src/display/editor/highlight.js

```diff
@@ -113,16 +116,20 @@ class HighlightEditor extends AnnotationEditor {
   }
 
   #createFreeOutlines({ highlight, highlightId, clipPathId }) {
-    this.#highlightOutlines = highlight.getOutlines(
-      this._uiManager.direction === "ltr"
-    );
-    this.#id = highlightId;
-    this.#clipPathId = clipPathId;
-    const { x, y, width, height, lastPoint } = this.#highlightOutlines.box;
+    this.#highlightOutlines = highlight;
+    if (highlightId >= 0) {
+      this.#id = highlightId;
+      this.#clipPathId = clipPathId;
+    }
+    const { x, y, width, height, lastPoint } = highlight.box;
 
-    // We need to redraw the highlight because we change the coordinates to be
-    // in the box coordinate system.
-    this.parent.drawLayer.finalizeLine(this.#id, this.#highlightOutlines);
+    if (highlightId >= 0) {
+      // We need to redraw the highlight because we change the coordinates to be
+      // in the box coordinate system.
+      this.parent.drawLayer.finalizeLine(highlightId, highlight);
+    } else {
+      this.parent?.drawLayer.updateLine(this.#id, highlight);
+    }
```

Comment:
Can we combine this block with the existing `if (highlightId >= 0) {`-block above?
E.g. something like the following:
```javascript
    const { x, y, width, height, lastPoint } = highlight.box;

    if (highlightId >= 0) {
      this.#id = highlightId;
      this.#clipPathId = clipPathId;

      // We need to redraw the highlight because we change the coordinates to be
      // in the box coordinate system.
      this.parent.drawLayer.finalizeLine(highlightId, highlight);
    } else {
      this.parent?.drawLayer.updateLine(this.#id, highlight);
    }
```